### DISCUSSION TITLE
Is it possible to add the caption as parameter (Custom zoom modal content)

### DISCUSSION THREAD

**Discussion by DeathdragonD** [rpearce/react-medium-image-zoom#398] [Q&A]
Hi,

I tried the example with the Custom zoom modal content and it seems that it's currently not possible to add a new parameter to the CustomZoomContent object, e.g. to show a caption for different images in a gallery.

It seems that if you add a prop manually, its getting deleted. Is there are way to achieve the described functionality in this version?

If its already possible, it would help to improve the example by adding it:
[Custom zoom modal content](https://rpearce.github.io/react-medium-image-zoom/?path=/story/img--modal-figure-caption)

If its currently not supported to do that, adding a feature to support that would be greatly appreciated.

Thanks!

**Comment by rpearce** [comment]
@DeathdragonD Are you using the same `<Zoom>` component for multiple images and captions?

I'll see what providing a different parameter to the custom zoom object does (or doesn't do) and check back.

**Reply by rpearce** [reply]
@DeathdragonD I was doing other things and forgot about this. I'll see what I can do

**Reply by DeathdragonD** [reply]
Hi, yes I built a gallery that uses the same Zoom Component via going through the images via a for-loop and I use the CustomZoomComponent to support downloading of images, if I could add custom parameters to the object I could provide different captions for each Image. 

If I remember correctly it may could even be used to cycle trough the images (e.g. have a next, prev picture function) if I add a ref to previous and next picture via a custom parameter?

**Reply by DeathdragonD** [reply]
Hi, is there anything new on this issue/feature?

Thanks!

**Reply by rpearce** [reply]
We had a second child, and I honestly forgot all about this. Can you make a feature request issue here so I don't lose it again? https://github.com/rpearce/react-medium-image-zoom/issues/new
